src/main/scala/org/hammerlab/guacamole/readsets/iterator/ContigCoverageIterator.scala

```diff
@@ -53,11 +52,28 @@ case class ContigCoverageIterator(halfWindowSize: Int,
       else
         lastStart = start
 
-      ends.enqueue(end)
-      numAdded += 1
+      if (end + halfWindowSize > locus) {
```

Comment:
I think I follow this change but not 100%, could you add some comments here about what should be happening?